corehq/apps/registration/forms.py

```diff
@@ -632,9 +644,20 @@ def clean(self):
             raise forms.ValidationError(error)
         return cleaned_data
 
-    def _initialize_tableau_fields(self, data, domain):
+    def _initialize_tableau_fields(self, data, domain, invitation=None):
```

Comment:
EditWebUser view has a permission check to set TableauUserForm fields as read only (see [here](https://github.com/dimagi/commcare-hq/blob/2471e5b765fa9ada13f3d246e172f14f369caec4/corehq/apps/users/views/__init__.py#L358))

Now that the Invite form can be edited, the `readonly` check also needs to be done. I think [this](https://github.com/dimagi/commcare-hq/blob/33159f52758a9cc37b356648b81ea17e5d0e0059/corehq/apps/users/forms.py#L1760-L1762) can be moved to the BaseTableauUserForm?

Reply:
again, I think the main use case for this feature is to allow the user who made the invitation to go back and fix any mistakes they might've made in the initial invitation and making those fields readonly after the initial invite was sent out defeats that purpose. I think generally we can treat invites and web user editing as two different things where the former has less risks associated with it because the user either doesn't exist yet or at least doesn't have any sort of access to the domain and its contents yet.

Reply:
I'm suggesting read only is set only for users who don't have permission to edit tableau config. (When creating the invite, the user wouldn't have been allowed to set the tableau config if the user does not have edit tableau permission).

Also, I think permissions permissions for editing invites should work the same way as editing existing web users because others, not just the person who created the invite, can access and edit it.